js/main.js

```diff
@@ -21,21 +44,34 @@ function saveEntry(event) {
   };
   dataObject.entryId = data.nextEntryId;
   data.nextEntryId++;
-  data.entries.unshift(dataObject);
-  $image.setAttribute('src', 'images/placeholder-image-square.jpg');
-  var addNew = createDom(dataObject);
-  $parent.prepend(addNew);
-  $noEntry.className = 'hidden';
-  $form.reset();
   switchView('entries');
+  if ($h1.textContent === 'New Entry') {
+    $image.setAttribute('src', 'images/placeholder-image-square.jpg');
+    var addNew = createDom(dataObject);
+    $parent.prepend(addNew);
+    data.entries.unshift(dataObject);
+    $noEntry.className = 'hidden';
+    $form.reset();
+  } else {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        data.editing.title = $form.elements.title.value;
+        data.editing.photo = $form.elements.photo.value;
+        data.editing.notes = $form.elements.notes.value;
+        var addEdit = createDom(data.editing);
+        var $mainList = document.querySelector('.mainlist');
+        $parent.prepend(addEdit);
+        $parent.removeChild($mainList);
+        switchView('entries');
+      }
```

Comment:
I'm not sure if this is doing the behavior expected for this feature 🤔 

When an edited entry is submitted, you should replace that object in `data.entries` with the newly edited object, and also replace the previous DOM tree for that entry with a DOM tree for the new edited entry, it should not be at the top of the list since it is not a new entry.